### ISSUE TITLE
Batch size < GPU number when training with Trainer and deepspeed.

### ISSUE THREAD

**Issue by zhaowei-wang98** [huggingface/transformers#16750]
# 🚀 Feature request
Hi, I am fine-tuning T5-11b using Trainer with deepspeed feature. I use the deepspeed zero3 stage to split T5-11b and its gradients to different GPUs. However, when I try to use multi GPUs, I found that the argument `per_device_train_batch_size` must be an integer, which means it at least is 1. So when I use more GPUs, the batch size must increase at the same time, which will cost must more GPU memory. Thus, it turns out that I can't fine-tune T5-11b with 2, 4 or 8 A100 (40G) GPUs. So, in general, deepspeed feature doesn't solve the memory issue if the model's size is similar to or larger than the memory of one of the multi GPUs.
So, I request the feature that batch size is smaller than the number of GPUs like train batch size of 2 on 4 GPUs.
here is the link to the  argument`per_device_train_batch_size`
https://github.com/huggingface/transformers/blob/main/src/transformers/training_args.py#L122

## Motivation

Fine-tune an extremely large model with a few small memory GPUs.

## Your contribution
I think the deepspeed package supports this feature already. So, adding this feature to Trainer is not hard.

**Comment by LysandreJik** [comment]
cc @sgugger

**Comment by sgugger** [comment]
cc @stas00 for DeepSpeed ;-)

**Comment by stas00** [comment]
@zhaowei-wang98, could you please try again explaining what is the problem that you're running into? Please show the actual command line / config you're running as I have a hard time understanding your Issue.

> So when I use more GPUs, the batch size must increase at the same time, which will cost must more GPU memory

There is no such requirement. 

In general there is no problem running T5-11B on A100 (40GB) w/ Deepspeed ZeRO-3 - or at least it worked last time I run it - It was done already more than a year ago, perhaps have a look at this old thread? https://github.com/huggingface/transformers/issues/9996 and then if you're still stuck tell us more details about your particular setup?

**Comment by stas00** [comment]
OK, but I still can't help you since I don't know how to reproduce your issue as you gave no specific instructions to do so, nor you shared anything about your setup other than the type of GPUs.

But I can probably do some guessing:

To train t5-11b you need at least `11*18=200`GB of memory just for the optim/grads/weights (I assume mixed precision) plus memory for activations and temps, so let's say roughly 240GB. 

With 40GB GPUs, that means at least 6 gpus. So it should be possible to load it on 8x 40GB gpus.

Also I recommend for you to switch to sharded version of t5-11b which I have just [made](https://github.com/huggingface/transformers/issues/16884), by passing to the trainer:  `--model_name_or_path t5-11b --model_revision sharded`

Because if you don't shard you would need 44GB per process, just to load the checkpoint (deepspeed shards it directly to gpus). And with 8 gpus you'd need 352GB of CPU memory just to load 8 checkpoints concurrently.

I think with 10GB shards some 100GB should be enough, but then there are extras to copy things and temps.

I'd be very happy to help you sort it out, but you need to help me first. To continue please be very specific:

1. here is my hardware setup
2. here is my software setup
3. here is my command line (using public data)
4. here is the traceback

Thank you!
